### ISSUE TITLE
Use qualids instead of idents to refer to Declare Reduction names, scopes, hint databases

### ISSUE THREAD

**Issue by jfehrle** [coq/coq#12063]
For consistency, we should use qualids to identify Declare Reduction names, scopes hint databases, etc. which are now IIUC always global.  This may raise compatibility issues that we'd have to address.

Opinions on the merit and/or difficulty of doing this?

See https://github.com/coq/coq/pull/11961#discussion_r405470583

**Comment by ppedrot** [comment]
+10.000. Technically these should be kernel names rather than qualified identifiers, but that's about the same. Also, this should be a high-priority task for some of the recently introduced features so as not to let the code relying on the current implementation grow. Furthermore, we should forbid any new feature that introduces superglobal names.

**Comment by Zimmi48** [comment]
>+10.000

I think you meant to say "+10,000"

**Comment by Zimmi48** [comment]
Starting with `Declare Reduction` might be much simpler than attempting the same with `HintDb`.

**Comment by bcip** [comment]
This is what I am thinking about recently, and it is surprising to find you are discussing about this just a few weeks ago. I provide some comments anyway, although I suppose you already have all of them in mind as you are Coq experts.

Making these names qualified will certainly benefits large-scale projects, that users will not need to concerned about using the same name for HintDb in another part of project. For this purpose, I think we need to make it mandatary to explicitly create hint databases before adding hints, otherwise it will be ambiguous whether to add to a new HintDb in current module or to an existing HintDb imported from another module. And, as far as I know, there is not a way to create a rewrite database.

**Comment by Zimmi48** [comment]
@bcip You are correct. We should add a command to create a rewrite database and deprecate implicit creation of hint databases, like we did already for scopes.

**Comment by Zimmi48** [comment]
These first steps are likely not so hard to do BTW.
